AutoEncoderTest_Gesamt.py

```diff
@@ -0,0 +1,599 @@
+"""
+Code ist im Rahmen der Bachelorarbeit von Annika Kies am FZI Karlsruhe mit dem Titel
+"Entwicklung und Analyse von Auto-Encodern für intelligente Agenten zum Erlernen von Atari-Spielen"
+entstanden.
+
+Code dient zum Testen eines Auto-Encoders.
+Ein trainierter Auto-Encoder kann mit einem Testdatensatz geprüft werden.
+Ermittelt wird der Median Loss, beste Loss und schlechteste Loss im gesamten Testdatensatz.
+
+Implementiert sind vier Auto-Encoder Architekturen.
+AutoEncoder1 orientiert sich an:
+Paper: Ha and Schmidhuber, "World Models", 2018. https://doi.org/10.5281/zenodo.1207631.
+https://github.com/ctallec/world-models
+"""
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from matplotlib import pyplot as plt
+from PIL import Image
+import numpy as np
+from numpy import load
+from sklearn.model_selection import train_test_split
+from torch.utils.data import TensorDataset
+from numpy import asarray
+from visdom import Visdom
+import utils
+import cv2
+import statistics
+
+
+
```

Comment:
Alle AutoEncoder Implementierungen sollten nur in einer Datei implementiert sein und nicht dreimal. Am Besten wäre es die auszulagern und dann dort wo man sie brauch zu importieren